src/main/java/pl/touk/sputnik/connector/ConnectorFacade.java

```diff
@@ -12,5 +15,14 @@
     @NotNull
     List<ReviewFile> listFiles();
 
+    /**
+     * Validates if given options are supported by selected connector. If not it throws an exception with information
+     * which option won't be supported and shall be changed.
+     * 
+     * @throws GeneralOptionNotSupportedException
+     *             if configuration is invalid
+     */
+    void supports(Configuration configuration) throws GeneralOptionNotSupportedException;
```

Comment:
That was your suggestion as mentioned above but I agree that validate tells more what is the purpose of the method.